### PR TITLE
[FSR] 64213 - Review Navigation addition

### PR DESCRIPTION
**Pull request by Dr-Pongo**
## Summary
This PR adds in review page navigation for the FSR which involves adding titles & buttons inside each accordion on the final review page that takes the user back into the flow of the form where they edit every page in that chapter and then return to the review page at the end of that chapter. We had a number of issues with in line editing on the final review page, and it's not the most accessible option, so going back into the form seemed reasonable enough. We included an alert on the first page of each chapter to inform the user of the flow and give them the option to return to the review page. At he definitive end of each chapter we have updated the text of the "continue" button to read "return to review", butttt we can't update the button text if they make changes to a page that will force it to be the end of the chapter (see `Income chapter` / `Asset chapter` 's  "End of Chapter" drop downs below) so the continue button text will still read "Continue".

## Related issue(s)

- Original ticket
department-of-veterans-affairs/va.gov-team#64213

- Testing ticket 
department-of-veterans-affairs/va.gov-team#66330

- Previous PR establishing custom pages to allow for review navigation 
 #26272

## Testing done
Manual testing (in progress)

## Screenshots
---
### Veteran information chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head (first page in chapter on reveiw)</summary>
image
</details>
<details>
<summary>End of chapter page(s)</summary>

**NOTE:** Count will be the final page if it is zero

| Dependent count | Dependent Ages |
| --- | --- |
| image | image |

</details>

---
### Income chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head (first page in chapter on reveiw)</summary>
image
</details>
<details>
<summary>End of chapter page(s)</summary>

**NOTE:** If there is no spouse listed the final pages of this chapter will be veteran other income related. If there is a spouse listed the final pages will be spouse other income related

| Veteran other income checklist | Veteran other income summary |
| --- | --- |
| image | image |

| Spouse other income checklist | Spouse other income summary |
| --- | --- |
| image | image |

</details>

---
### Asset Chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head(s) (first page in chapter on reveiw)</summary>

**NOTE:** Cash on hand will be first page in chapter for streamlined flow

| Monetary Assets | Cash on hand |
| --- | --- |
| image | image |
</details>
<details>
<summary>End of chapter page(s)</summary>

**NOTE:** If no assets are selected the checklist will be the final page in the chapter

| Other assets checklist | Other assets summary |
| --- | --- |
| image | image |

</details>

---
### Expenses Chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head (first page in chapter on reveiw)</summary>
image
</details>
<details>
<summary>End of chapter page(s)</summary>

**NOTE:** If no expenses are selected the checklist will be the final page in the chapter

| Other expenses checklist | Other expenses summary |
| --- | --- |
| image | image |

</details>

---
### Resolution options chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head (first page in chapter on reveiw)</summary>
image
</details>
<details>
<summary>End of chapter page</summary>
image
</details>

---
### Bakruptcy Chapter
<details>
<summary>Review page</summary>
image
</details>
<details>
<summary>Chapter Head (first page in chapter on reveiw)</summary>
image
</details>
<details>
<summary>End of chapter page(s)</summary>
</details>

---
### Hook - useDetectFieldChanges 
<details>
<summary>Review Navigation Hook</summary>

#### Core functionality:
The useDetectFieldChanges hook is like a vigilant guard for our form's "Review" button. It watches over the form data, particularly focusing on changes related to spouse details and the form's streamlined state.

- Monitors spouse-related fields and form streamlined state.
- Determines visibility of "Review" button based on changes.
- Uses previous form data for comparison without extra renders.
- Smart helpers assess if the form or spouse details change.
- Reacts only to changes that affect the "Review" button's need to be visible.
- In app.jsx, it triggers form state updates cautiously to avoid infinite loops.
</details>

---

## What areas of the site does it impact?
FSR 5655
*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by OptionSelect — APPROVED** [review]
LGTM is an understatement. Great work bud! 🚀

**Comment by Dr-Pongo** [comment]
Some known issues to be addressed:

- Contact information page is setting focus to "Need Help?" h2 instead of "contact information" header
- Alert showing up after page header (should be before)
  - Work history
  - Veteran information
  - Bankruptcy

**Review by powellkerry — APPROVED** [review]

**Comment by powellkerry** [comment]
Looks like a change was made to an app with a flaky test. This will need to be addressed before it will allow you to merge: https://github.com/department-of-veterans-affairs/vets-website/actions/runs/6816750515/job/18539783614?pr=26509

**Review by amponce — APPROVED** [review]
Solid work! looks like we're ready to ship

File: src/applications/financial-status-report/components/shared/ButtonGroup.jsx [hunk 1]
```diff
@@ -15,39 +15,54 @@ import PropTypes from 'prop-types';
...
+            <button
+              type={button.type === 'submit' ? 'submit' : 'button'}
+              className={
+                button.secondary ? 'usa-button-secondary' : 'usa-button-primary'
+              }
+              onClick={button.onClick}
+              disabled={button.disabled}
+            >
+              {button.iconLeft && (
+                <i aria-hidden="true" className="fa fa-angles-left" />
```
[the remark this answers is not in the document]
Comment (Dr-Pongo) [reply src/applications/financial-status-report/components/shared/ButtonGroup.jsx:56]: `aria-hidden="true"` applied, and context is given

**Review by powellkerry — APPROVED** [review]

**Pull request merged**